pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/KitConfigurationDao.java

```diff
@@ -96,59 +114,65 @@ default long addCountryRule(long configId, String countryCode) {
-        String guid = jdbiUmbrellaStudy.findGuidByStudyId(kitConfigurationDto.getStudyId());
+        String guid = kitConfigurationDto.getStudyGuid();
 
         return new KitConfiguration(kitConfigurationDto.getId(), numKits, kitType, guid, kitRules);
     }
 
-    default Collection<KitRule> getTypedRulesForConfigurationId(Handle handle, Long configurationId) {
-        Collection<KitRule> kitRules = new ArrayList<>();
-
-        JdbiKitRules jdbiKitRules = handle.attach(JdbiKitRules.class);
-        JdbiExpression jdbiExpression = handle.attach(JdbiExpression.class);
-        JdbiCountry jdbiCountry = handle.attach(JdbiCountry.class);
-
-        Collection<KitRuleDto> kitRuleDtos = getUntypedRulesForConfigurationId(configurationId, jdbiKitRules);
-
-        for (KitRuleDto kitRuleDto : kitRuleDtos) {
-            switch (kitRuleDto.getKitRuleType()) {
+    @SqlQuery("select kckr.kit_configuration_id,"
+            + "       kr.kit_rule_id,"
+            + "       krt.kit_rule_type_code as kit_rule_type,"
+            + "       pe.expression_text as pex_expression,"
+            + "       (select country_code from country where country_id = kc.country_id) as country_code,"
+            + "       (select group_concat(zip_code separator ',') from kit_zip_code where kit_rule_id = kr.kit_rule_id) as zip_codes"
+            + "  from kit_configuration__kit_rule as kckr"
+            + "  join kit_rule as kr on kr.kit_rule_id = kckr.kit_rule_id"
+            + "  join kit_rule_type as krt on krt.kit_rule_type_id = kr.kit_rule_type_id"
+            + "  left join kit_pex_rule as kp on kp.kit_rule_id = kr.kit_rule_id"
+            + "  left join expression as pe on pe.expression_id = kp.expression_id"
+            + "  left join kit_country_rule as kc on kc.kit_rule_id = kr.kit_rule_id"
+            + " where kckr.kit_configuration_id = :configId")
+    @UseRowReducer(KitRuleReducer.class)
+    List<KitRule> findRulesByConfigId(@Bind("configId") long configId);
+
+    class KitRuleReducer implements LinkedHashMapRowReducer<Long, KitRule> {
+        @Override
+        public void accumulate(Map<Long, KitRule> container, RowView view) {
+            long ruleId = view.getColumn("kit_rule_id", Long.class);
+            KitRuleType type = KitRuleType.valueOf(view.getColumn("kit_rule_type", String.class));
+            KitRule rule;
+            switch (type) {
                 case PEX:
-                    KitPexRuleDto kitPexRuleDto = jdbiKitRules.getKitPexRuleById(kitRuleDto.getId()).get();
-                    PexInterpreter pexInterpreter = new TreeWalkInterpreter();
-                    String expression = jdbiExpression.getExpressionById(kitPexRuleDto.getExpressionId());
-                    kitRules.add(new KitPexRule(pexInterpreter, expression));
+                    String expression = view.getColumn("pex_expression", String.class);
+                    rule = new KitPexRule(ruleId, new TreeWalkInterpreter(), expression);
```

Comment:
Don't like how we have to `new` up a new interpreter for each one, but refactoring will require a lot more changes than what I can afford now. Most kit configurations only have one of these kit pex rules, so should be fine for now.